### PR TITLE
feat: added Braze plugin

### PR DESCRIPTION
**Pull request by falconandy**

### CODE REVIEW

File: packages/plugin-braze/tests/test_braze.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+        p.track("user-2", Event('event-1', Properties(item1='value1', item2=1)))
+        time.sleep(0.1)
+        requests = _get_cleaned_requests(httpserver)
+        assert requests == []
+
+        p.track("user-2", Event('event-2', Properties(item1='value2', item2=2)))
+        time.sleep(0.8)
+        requests = _get_cleaned_requests(httpserver)
+        assert requests == [
```
Comment (justin-iteratively) [inline packages/plugin-braze/tests/test_braze.py:41]: Thank you @falconandy for adding tests. To make this more manageable could you please break this down into multiple independent cases. Ideally each test would [follow AAA and the naming conventions outlined here](https://bitbucket.org/seasyd/analytics_backend/src/master/test/). E.g.

```
def test_braze(httpserver: HTTPServer):
    time_short = 0.1
    time_max = 999
    event_1 = Event('event-1', Properties(item1='value1', item2=1))
    event_2 = Event('event-2', Properties(item1='value2', item2=2))

    def before_each:
       httpserver.expect_request(re.compile('/users/track')).respond_with_data()
       base_url=httpserver.url_for('')
       
    def Id_BrazePlugin_IsBraze(httpserver: HTTPServer):
        p = BrazePlugin('My-Key', BrazeOptions(base_url=base_url)
        assert p.id() == 'braze'
        
    def Track_Immediate_MakesRequest(httpserver: HTTPServer):
        p = BrazePlugin('My-Key', BrazeOptions(
            base_url=base_url,
            flush_queue_size=1,
            flush_interval=timedelta(seconds=time_short),
        ))
        p.load()
        p.track("user-1", event_1)
        time.sleep(time_short * 2)

        assert p.id() == 'braze'
        
    def Flush_QueueSizeReached_MakesRequest:
        p = BrazePlugin('My-Key', BrazeOptions(
            base_url=base_url,
            flush_queue_size=3,
            flush_interval=timedelta(seconds=time_max),
        ))
        p.load()
        p.identify("user-1", Properties(item1='value1', item2=2))
        time.sleep(time_short)
        p.track("user-2", event_1)
        time.sleep(time_short)
        p.track("user-2", event_2)
        time.sleep(time_short)
        requests = _get_cleaned_requests(httpserver)
        assert requests == [{...}];

    def Flush_FlushIntervalReached_MakesRequest:
       ...

    def Shutdown_WithPendingEvents_FlushesAndMakesRequests:
       ....
```

File: packages/plugin-braze/itly_plugin_braze/_braze_client.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        atexit.register(self.shutdown)
+        self._consumer.start()
+
+    def identify(self, user_id: str, properties: Optional[Dict[str, Any]]) -> None:
+        data = self._to_braze_properties(properties)
+        data["external_id"] = user_id
+        self._enqueue(AsyncConsumerMessage("", {"attributes": data}))
+
+    def track(self, user_id: str, event_name: str, properties: Optional[Dict[str, Any]]) -> None:
+        data = self._to_braze_properties(properties)
```
Comment (justin-iteratively) [inline packages/plugin-braze/itly_plugin_braze/_braze_client.py:42]: I think Event properties need to use the key "properties" - https://www.braze.com/docs/api/objects_filters/event_object/#event-property-persistence

File: packages/plugin-braze/itly_plugin_braze/_braze_plugin.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+from datetime import timedelta
+from typing import NamedTuple, Optional
+
+from itly_sdk import Plugin, Properties, Event, PluginLoadOptions, Logger
+from ._braze_client import BrazeClient
+
+
+class BrazeOptions(NamedTuple):
+    base_url: str
+    flush_queue_size: int = 75
```
Comment (justin-iteratively) [inline packages/plugin-braze/itly_plugin_braze/_braze_plugin.py:10]: I see in the docs 75 is the maximum number of events per request. I think we should enforce the max in our implementation and set the default queue size to something lower than max. https://www.braze.com/docs/api/basics/#api-limits

File: packages/plugin-braze/itly_plugin_braze/_braze_plugin.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            on_error=self._on_error,
+            flush_queue_size=self._options.flush_queue_size,
+            flush_interval=self._options.flush_interval,
+            request_timeout=self._options.request_timeout,
+        )
+        self._logger = options.logger
+
+    def identify(self, user_id: str, properties: Optional[Properties]) -> None:
+        assert self._client is not None
+        self._client.identify(
```
Comment (justin-iteratively) [inline packages/plugin-braze/itly_plugin_braze/_braze_plugin.py:38]: Can you please add some logging for the Plugin methods

**Review by justin-iteratively — CHANGES_REQUESTED** [review]

File: packages/plugin-braze/tests/test_braze.py (hunk 1 above)
Comment (falconandy) [reply packages/plugin-braze/tests/test_braze.py]: Added more tests. Not sure about naming.

File: packages/plugin-braze/itly_plugin_braze/_braze_client.py (hunk 2 above)
Comment (falconandy) [reply packages/plugin-braze/itly_plugin_braze/_braze_client.py]: I've copied the implementation from JS version. Fixed.

File: packages/plugin-braze/itly_plugin_braze/_braze_plugin.py (hunk 3 above)
Comment (falconandy) [reply packages/plugin-braze/itly_plugin_braze/_braze_plugin.py]: 25?

File: packages/plugin-braze/itly_plugin_braze/_braze_plugin.py (hunk 4 above)
Comment (falconandy) [reply packages/plugin-braze/itly_plugin_braze/_braze_plugin.py:46]: Added

File: packages/plugin-braze/tests/test_braze.py (hunk 1 above)
Comment (justin-iteratively) [reply packages/plugin-braze/tests/test_braze.py]: Tests look great. Thanks.

**Review by justin-iteratively — APPROVED** [review]

**Pull request merged**